### ISSUE TITLE
deephaven2 unit test failure in nightlies

### ISSUE THREAD

**Issue by rcaudy** [deephaven/deephaven-core#1910]
```
  ======================================================================
  FAIL [0.004s]: test_datetime_at_midnight (test_time.DateT
  imeUtilsTestCase)
  ----------------------------------------------------------------------
  Traceback (most recent call last):
    File "/python/tests/test_time.py", line 68, in test_datetime_at_midnight
      self.assertGreaterEqual(diff_nanos(mid_night_time_ny, mid_night_time_pt), 0)
  AssertionError: -75600000000000 not greater than or equal to 0
```

**Comment by rcaudy** [comment]
-75600000000000 nanos is -21 hours
It looks like this ran after midnight ET and before midnight PT.

Test code:
```
    def test_datetime_at_midnight(self):
        dt = now()
        mid_night_time_ny = datetime_at_midnight(dt, TimeZone.NY)
        mid_night_time_pt = datetime_at_midnight(dt, TimeZone.PT)
        self.assertGreaterEqual(diff_nanos(mid_night_time_ny, mid_night_time_pt), 0)
```

This isn't a very good test, in general; we can be more specific in our expectations.
I think it makes more sense to have a set of input datetimes, and expected midnight datetimes in multiple timezones, including on dates with to/from DST transitions.

**Comment by jmao-denver** [comment]
fixed by #1812
